submissions/src/main/kotlin/gov/cdc/prime/reportstream/submissions/controllers/SubmissionController.kt

```diff
@@ -0,0 +1,185 @@
+     * to Azure Blob Storage, queued for further processing, and its metadata is stored in Azure Table Storage.
+     * A custom event is also tracked in Application Insights for monitoring.
+     *
+     * @param headers the HTTP headers of the request
+     * @param contentType the content type of the report (must be "application/hl7-v2" or "application/fhir+ndjson")
+     * @param clientId the ID of the client submitting the report. Should represent org.senderName
+     * @param data the report data
+     * @return a ResponseEntity containing the reportID, status, and timestamp
+     */
+    @PostMapping("/api/v1/reports", consumes = ["application/hl7-v2", "application/fhir+ndjson"])
+    fun submitReport(
+        @RequestHeader headers: Map<String, String>,
+        @RequestHeader("Content-Type") contentType: String,
+        @RequestHeader(value = "client_id", required = false) clientId: String?,
+        @RequestBody data: String,
+    ): ResponseEntity<*> {
+        val reportId = UUID.randomUUID()
+        val reportReceivedTime = OffsetDateTime.now()
+        val status = "Received"
+        logger.info("Received report submission: reportId=$reportId, contentType=$contentType, clientId=$clientId")
+
+        try {
+
+            val headerValidationResult = validateHeaders(clientId)
+            if (headerValidationResult != null) {
+                logger.warn("Header validation failed: $headerValidationResult")
+                return headerValidationResult
+            }
+
+            // Convert data to ByteArray
+            val dataByteArray = data.toByteArray()
+            logger.debug("Converted report data to ByteArray")
+
+            // Upload to blob storage
+            val blobContainerClient = blobServiceClient.getBlobContainerClient(containerName)
+            val blobClient = blobContainerClient.getBlobClient(formBlobName(reportId, headers))
+            blobClient.upload(dataByteArray.inputStream(), dataByteArray.size.toLong())
+            logger.info("Uploaded report to blob storage: blobUrl=${blobClient.blobUrl}")
+
+            // Create the message for the queue
+            val message = mapOf(
+                "reportId" to reportId.toString(),
+                "blobUrl" to blobClient.blobUrl,
+                "headers" to filterHeaders(headers)
+            )
+            val objectMapper = jacksonObjectMapper().registerModule(JavaTimeModule())
+            val messageString = objectMapper.writeValueAsString(message)
+            logger.debug("Created message for queue")
+
+            // Upload to Queue
+            val queueClient = queueServiceClient.getQueueClient(queueName)
+            queueClient.createIfNotExists()
+            queueClient.sendMessage(messageString)
+            logger.info("Sent message to queue: queueName=$queueName")
+
+            // Insert into Table
+            // TableEntity() sets PartitionKey and RowKey. Both are required by azure and combine to create the PK
```

Comment:
What do you think about using the reportReceivedTime as the partition key?

Reply:
yeah that's a bit of an improvement, added it